### PR TITLE
#3721 - PT Assessment - Books and Supplies Limit

### PR DESCRIPTION
**Pull request by andrepestana-aot**
- Added `limitWeeklyBooksAndSupplies` to `dmnPartTimeProgramYearMaximums`:
image

- Changed calculatedDataProgramRelatedCosts to check the minimum between `offeringProgramRelatedCosts` and `dmnPartTimeProgramYearMaximums.limitWeeklyBooksAndSupplies * offeringWeeks`:
image

- Added offering delivery "blended" to allow  `calculatedDataWeeklyMinTransport` to receive `dmnPartTimeProgramYearMaximums.limitTransportationAllowance`:
image

- Added offering delivery "blended" to allow  `inputDataRemainingTransportWeeks` to receive the value of `offeringWeeks - calculatedDataAdditionalTransportWeeks`
image

- Added offering delivery "blended" to allow  `calculatedDataTotalTransportationAllowance` to receive the value of `offeringWeeks * dmnPartTimeProgramYearMaximums.limitTransportationAllowance`
image

- Added tests to check program related costs impacted by books and supply costs;
- Added tests to check transportation costs when the offering delivery is blended;
- Added `calculatedDataTotalAssessedNeed` to [assessment.model.ts](https://github.com/bcgov/SIMS/pull/3771/files#diff-4cbd9602bdc302ca9d2093481096678ec9bdeada644a88152a74124fa413c49f) as this variable has a typo for FT and it is correct for PT. Added a TODO to be done in another ticket;
- Removed the python3 and make versions to fix a problem with local forms container (not related to this ticket).

### CODE REVIEW

File: sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-decisions.dmn [hunk 1]
```diff
@@ -1,5 +1,5 @@
...
+<definitions xmlns="https://www.omg.org/spec/DMN/20191111/MODEL/" xmlns:dmndi="https://www.omg.org/spec/DMN/20191111/DMNDI/" xmlns:dc="http://www.omg.org/spec/DMN/20180521/DC/" xmlns:biodi="http://bpmn.io/schema/dmn/biodi/2.0" xmlns:modeler="http://camunda.org/schema/modeler/1.0" xmlns:camunda="http://camunda.org/schema/1.0/dmn" id="dmnPartTimeAssessmentDecisions" name="dmnPartTimeAssessmentDecisions" namespace="http://camunda.org/schema/1.0/dmn" exporter="Camunda Modeler" exporterVersion="5.27.0" modeler:executionPlatform="Camunda Cloud" modeler:executionPlatformVersion="8.1.0" camunda:diagramRelationId="1a575ead-1e6d-468c-901a-766a35569b38">
```
Comment (lewischen-aot) [inline sources/packages/backend/workflow/src/workflow-definitions/parttime-assessment-decisions.dmn:2]: Thanks for updating the exporter version ❤️

File: sources/packages/forms/Dockerfile.dev [hunk 2]
```diff
@@ -8,8 +8,8 @@ ARG FORMIO_SOURCE_REPO_TAG
 # Note: using pinned versions to ensure immutable build environment.
 RUN apk update && \
   apk upgrade && \
-  apk add make=4.4.1-r2 && \
-  apk add python3=3.12.6-r0 && \
+  apk add make && \
```
Comment (andrewsignori-aot) [inline sources/packages/forms/Dockerfile.dev:11]: Thanks for taking care of this. I would suggest removing the version from the other packages also (lines 13 and 14).

**Review by andrewsignori-aot — APPROVED** [review]
Great work, looks good 👍

File: sources/packages/backend/workflow/test/2024-2025/parttime-assessment/costs/parttime-assessment-costs-program-related-costs.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  ZeebeMockedClient,
+  createFakeConsolidatedPartTimeData,
+  executePartTimeAssessmentForProgramYear,
+} from "../../../test-utils";
+
+describe(`E2E Test Workflow parttime-assessment-${PROGRAM_YEAR}-program-related-costs.`, () => {
+  it(
+    "Should use minimum (offering program related costs) " +
+      "when dmn limit for books and supplies * number of offering weeks has a higher value.",
+    async () => {
```
Comment (lewischen-aot) [inline sources/packages/backend/workflow/test/2024-2025/parttime-assessment/costs/parttime-assessment-costs-program-related-costs.e2e-spec.ts:12]: Minor suggestion. The test case description is great, and maybe we could change this to "Should get program related costs as offering program related costs when the DMN limit for books and supplies multiplied by offering weeks has the greater value."

**Pull request merged**